sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -61,6 +62,11 @@ private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
       // When neither spark.sql.warehouse.dir nor hive.metastore.warehouse.dir is set,
       // we will set hive.metastore.warehouse.dir to the default value of spark.sql.warehouse.dir.
       val sparkWarehouseDir = sparkContext.conf.get(WAREHOUSE_PATH)
+      logInfo(s"Setting hive.metastore.warehouse.dir ($hiveWarehouseDir) to the value of " +
+        s"${WAREHOUSE_PATH.key} ('$sparkWarehouseDir').")
+      if (hiveWarehouseDir != null) {
+        sparkContext.hadoopConfiguration.set("hive.metastore.warehouse.dir", sparkWarehouseDir)
+      }
       sparkContext.conf.set("hive.metastore.warehouse.dir", sparkWarehouseDir)
```

Comment:
I removed  `hive.metastore.warehouse.dir` in sparkConf above commit ,but some tests failed, so I revert this logic, let me dig it more.